push_dialog.py

```diff
@@ -87,35 +90,76 @@ def update_progress(self, sent, total):
         progress = float(sent) / total * 100
         self.progress_bar.setValue(progress)
 
-    def push_project(self):
+    def get_layer_ids_to_offline_convert(self, remote_layers, remote_save_mode):
+        layer_ids = []
+        if remote_save_mode == RemoteOptionsDialog.OFFLINE: 
+            for layer in remote_layers:
+                layer_ids.append(layer.id())
+
+        for layer in self.project_get_always_offline_layers():
+            layer_ids.append(layer.id())
+        return layer_ids
+
+
+    def push_project(self, remote_layers=None, remote_save_mode=None):
+
+        can_only_be_online_layers = self.project_get_always_online_layers()
+        if can_only_be_online_layers:
+            self.show_warning_about_layers_that_cant_work_offline(can_only_be_online_layers)
+
+        layer_ids = self.get_layer_ids_to_offline_convert(remote_layers, remote_save_mode)
+        # TODO: offline convert
+        # with rasters too?
+
+        if remote_save_mode == RemoteOptionsDialog.HYBRID:
+            self.set_hybrid_flag()
+        # TODO: show actual, more informative dialog with button
+        QtGui.QDesktopServices.openUrl(QtCore.QUrl.fromLocalFile(BASE_SAVE_LOCATION))
```

Comment:
@mbernasocchi Here we would have in mind to show another dialog that says "project has been exported to /tmp/, please copy it manually. <button to open folder>

What I am not sure is how to decide where to export the project (I have hardcoded /tmp/ for now)? Should this be customizable/settable via the UI? Because windows machines don't have a tmp folder